content/kb/troubleshooting-volume-readonly-or-io-error.md

```diff
@@ -0,0 +1,73 @@
+......
+time="2021-01-05T11:20:46Z" level=warning msg="Engine of volume dead unexpectedly, reattach the volume" accessMode=rwo controller=longhorn-volume frontend=blockdev node=shuo-cluster-0-worker-3 owner=shuo-cluster-0-worker-3 state=attached volume=pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c
+......
+time="2021-01-05T11:20:46Z" level=info msg="Event(v1.ObjectReference{Kind:\"Volume\", Namespace:\"longhorn-system\", Name:\"pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c\", UID:\"69bb0f94-da48-4d15-b861-add435f25d00\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"7466467\", FieldPath:\"\"}): type: 'Warning' reason: 'DetachedUnexpectly' Engine of volume pvc-342edde0-d3f4-47c6-abf6-bf8eeda3c32c dead unexpectedly, reattach the volume"
+```
+
+### Most common/Direct reason
+The mount point of the Longhorn volume becomes invalid once the Longhorn volume crashes unexpectedly. Then there is no way to read or write data in the Longhorn volume via the mount point.
+
+For Longhorn version earlier than v1.1.0, Longhorn will try to remount the volume automatically but it doesn't work well.
+
+Since Longhorn version v1.1.0, a new setting [`Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`](https://longhorn.io/docs/1.1.0/references/settings/#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) is introduced so that Longhorn will automatically delete the workload pod that is managed by a controller (e.g. deployment, statefulset, daemonset, etc...).
+
+#### Solution
+If the workload is a simple pod, you can delete and re-deploy the pod. Please make sure the related PVC or PV is not removed if the reclaim policy is not `Retain`. Otherwise, the Longhorn volume will be removed once the related PVC/PV is gone.
+
+If the workload pod belongs to Deployment/StatefulSet, you can restart the pod by scaling down then scaling up the workload replica.
+
+And for Longhorn v1.1.0 or higher version, you can enable setting `Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly`.
+
+### Root cause
+The key point is, what leads to the unexpected engine crash. Here are the possible reasons/checklist:
+
+1. CPU utilization on the node. If the CPU resource of the instance manager pods are limited, the engines insides the pod may get crashed. You can refer to [this doc](https://longhorn.io/docs/1.1.0/best-practices/#guaranteed-engine-cpu)  to see how to reserve proper amount of CPU resources for Longhorn instance manager pods.
+2. The network bandwidth is limited. Normally 1Gbps network will only able to serve 3 volumes if all of those volumes are running a high intensive workload.
+3. The network latency is relatively high. If there are multiple volumes r/w simultaneously on a node, it's better to guarantee that the latency is less than 20ms.
+4. A network interruption can result in all-replica disconnection then engine crash.
+5. The disk type or IO performance. Please avoid using HDD disks in Longhorn system.
```

Comment:
`The performance of the disk is too low to finish the request in time. We don't recommend using low IOPS disks (e.g. spinning disks) in the Longhorn system`.